app/controllers/books/comments_controller.rb

```diff
@@ -1,20 +1,35 @@
 # frozen_string_literal: true
 
-class Books::CommentsController < CommentsController
+class Books::CommentsController < ApplicationController
   before_action :set_book, only: %i[create destroy]
 
   def create
-    super(@book)
+    @comments = @book.comments
+    @new_comment = @comments.build(comment_params)
```

Comment:
```suggestion
    @comment = @book.comments.build(comment_params)
```

でもいいと思います。

newは不要ですね。
scaffoldで作ったコードも`@new_user = User.new`ではなく、`@user = User.new`ってなってますし。

あと、このコードは

```suggestion
    comment = @book.comments.build(comment_params)
```

でも動きます。というか、そっちの方がベターです。なぜでしょうか？

Reply:
変数名のnewの部分を削除しました。

>でも動きます。というか、そっちの方がベターです。なぜでしょうか？

仰るとおり、インスタンス変数にする必要はありません。修正漏れです。
昔のコードで`render`を使用していたため、インスタンス変数にする必要がありましたが、
今のコードは、`redirect`のみなのでローカル変数に修正すべきでした。